### PR TITLE
Symbol protection

### PR DESCRIPTION
**Pull request by WebFreak001**
Moves protection into DSymbol

Needed for dlang-community/DCD#598

### CODE REVIEW

File: src/dsymbol/scope_.d [hunk 1]
```diff
@@ -215,6 +216,11 @@ struct Scope
 		return getSymbolsByName(name);
 	}
 
+	bool hasSymbolRecursive(const(DSymbol)* symbol) const
```
Comment (WebFreak001) [inline src/dsymbol/scope_.d]: this is not strictly part of the protection changes, I added it for the use-case of after checking protection in DCD

**Comment by WebFreak001** [comment]
I don't think this is a very complex change, it just makes the protection available inside a `DSymbol`. Memory usage per `DSymbol` would maybe rise a little bit (or well, freeing SemanticSymbol won't free the few protection bytes anymore) but I don't think that would be that big of an issue? Before it was in `SemanticSymbol`. It was also the only property in there that was only related to the `DSymbol` instance, so I think it even makes sense from a semantic view to move it into `DSymbol` as it describes it.

@Basile-z what do you think?

**Comment by Basile-z** [comment]
I'm struggling on what the other IDEs do. Lazarus do show private members. I don't remember what Delphi did, and otherwise I don't know. I think that this change was proposed by @CyberShadow initially. In other words, I wont be a decision maker here.

**Comment by WebFreak001** [comment]
for the dsymbol PR here I don't think any of that matters, it simply adds information about the visibility level of a symbol. I think this should be merged and a decision on hiding private members should be made on the DCD PR.

**Comment by Basile-z** [comment]
tagged v0.7.0 since this modifies the API

**Pull request merged**